cloudflare/workers/backblaze.js

```diff
@@ -0,0 +1,58 @@
+// Improved the script from this blog https://www.grahn.io/posts/2020-02-08-s3-vs-b2-static-web-hosting/
+// Backblaze Url
+const baseURL = "${{ secrets.backblazeBucketUrl }}"
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event))
+})
+
+async function handleRequest(event) {
+  // only allow get requests
+  if (event.request.method !== 'GET') {
+    return new Response('Method not allowed', { status: 405 })
+  }
+  // return a cached response if we have one
+  const cache = caches.default
+  let cachedResponse = await cache.match(event.request)
+  if (cachedResponse) {
+    return cachedResponse
+  }
+  
+  const parsedUrl = new URL(event.request.url)
+  let path = parsedUrl.pathname
+  const lastSegment = path.substring(path.lastIndexOf('/'))
+  // insert index page when path has a trailing slash
+  if (path.endsWith('/')) {
+    path += 'index.html'
+  }
+
+  // fetch content from B2
+  const response = await b2Fetch(path)
+  // all is well, return the response
+  if (response.status < 400) {
+    event.waitUntil(cache.put(event.request, response.clone()))
```

Comment:
Nice, so it writes to the cache asynchronously but returns the response immediately. Hence `waitUntil` intead of `await`

Reply:
Yes, https://developers.cloudflare.com/workers/runtime-apis/fetch-event#waituntil